frontend/public/components/utils/triggers.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import { K8sResourceKind } from '../../module/k8s';
+import { ResourceLink } from './';
+
+export const Triggers: React.SFC<TriggersProps> = ({ resource }) => {
+  const triggers = _.get(resource, 'spec.triggers');
+  const getTriggerRefType = trigger => {
+    switch (trigger.type) {
+      case 'Bitbucket':
+        return 'bitbucket';
+      case 'Generic':
+        return 'generic';
+      case 'GitHub':
+        return 'github';
+      case 'GitLab':
+        return 'gitlab';
+      default:
+        return null;
+    }
+  };
+  const kubeAPIServerURL = (window as any).SERVER_FLAGS.kubeAPIServerURL;
+  const namespace = resource.metadata.namespace;
+  const buildConfigName = resource.metadata.name;
+  const getWebhookURL = triggerRefType => (triggerRefType === 'bitbucket' || triggerRefType === 'generic' || triggerRefType === 'github' || triggerRefType === 'gitlab')
+    ? `${kubeAPIServerURL}/oapi/v1/namespaces/${namespace}/buildconfigs/${buildConfigName}/webhooks/<secret>/${triggerRefType}`
+    : '-';
+  const getSecretReference = (trigger, triggerRefType) => ((triggerRefType === 'bitbucket' || triggerRefType === 'generic' || triggerRefType === 'github' || triggerRefType === 'gitlab') && trigger[triggerRefType].secretReference)
```

Comment:
I'd add a helper to check if something is a valid trigger to avoid repeating the logic here and above. If you create a `validTriggers` set per @alecmerdler, the check is simpler.